### PR TITLE
Use react-router for persisting state and remove Sinatra

### PR DESCRIPTION
**Pull request by jason-crowley**
This PR aims to fix issues with the form being in an inconsistent state when hitting the browser back button after validation. However, to prevent these browser inconsistencies, we had to handle routing through React by using react-router and removing Sinatra. In this PR we move the job of querying the validator wrapper and serving different pages into React-land. (Previously this was done in Sinatra.)

### CODE REVIEW

**Comment by okeefm** [comment]
@radamson / @jason-crowley should we delete all the Sinatra stuff in this PR, if we're no longer using it?

**Comment by jason-crowley** [comment]
That makes sense to me. So do I just delete everything I'm not using, including all of the Ruby code?

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "test": "jest",
     "test:watch": "jest --watch",
     "start": "webpack --config webpack.dev.js",
```
Comment (okeefm) [inline package.json:13]: There's probably not a real need for both `start` and `start:dev` without Sinatra, so you could probably remove this line.

Comment (okeefm) [reply package.json:13]:
```suggestion
```

Comment (jason-crowley) [reply package.json:13]: Ok I'll just rename the current `start:dev` to `start` then.

**Comment by jason-crowley** [comment]
> @radamson / @jason-crowley should we delete all the Sinatra stuff in this PR, if we're no longer using it?

Oh, and that reminds me that I should also update the documentation to remove the `bundle exec rackup` command

**Comment by jason-crowley** [comment]
Looking at the Ruby files that I deleted, I realize that I probably should have made a separate `HL7Validator` class like you had to make it easier to test as well as to separate the validation logic from the render logic. If I make that class, would that be best to put into another PR or should I try to do that in this PR?

**Comment by okeefm** [comment]
I think you'll have to change `4567` in this line of the README to `80`, since we changed the port the dockerfile listens on to `80` 

https://github.com/inferno-community/fhir-validator-app/pull/27/files#diff-04c6e90faac2675aa89e2176d2eec7d8L25

**Comment by jason-crowley** [comment]
Sorry I forgot to push the changes to the README! As for the environment variables, there's a commit in my load-ig-packages PR that implements that. It might be hard to move here, so is it alright if I keep it in the other PR?

**Pull request merged**